lib/winapi/fileapi.h

```diff
@@ -3,19 +3,45 @@
 
 #include <windef.h>
 #include <winbase.h>
+#include <winnt.h>
 
 #ifdef __cplusplus
 extern "C"
 {
 #endif
 
+#define CREATE_NEW 1
+#define CREATE_ALWAYS 2
+#define OPEN_EXISTING 3
+#define OPEN_ALWAYS 4
+#define TRUNCATE_EXISTING 5
+
+HANDLE CreateFileA (LPCSTR lpFileName, DWORD dwDesiredAccess, DWORD dwShareMode, LPSECURITY_ATTRIBUTES lpSecurityAttributes, DWORD dwCreationDisposition, DWORD dwFlagsAndAttributes, HANDLE hTemplateFile);
+BOOL ReadFile (HANDLE hFile, LPVOID lpBuffer, DWORD nNumberOfBytesToRead, LPDWORD lpNumberOfBytesRead, LPOVERLAPPED lpOverlapped);
+BOOL WriteFile (HANDLE hFile, LPCVOID lpBuffer, DWORD nNumberOfBytesToWrite, LPDWORD lpNumberOfBytesWritten, LPOVERLAPPED lpOverlapped);
+DWORD SetFilePointer (HANDLE hFile, LONG lDistanceToMove, PLONG lpDistanceToMoveHigh, DWORD dwMoveMethod);
+BOOL SetFilePointerEx (HANDLE hFile, LARGE_INTEGER liDistanceToMove, PLARGE_INTEGER lpNewFilePointer, DWORD dwMoveMethod);
+
+DWORD GetFileSize (HANDLE hFile, LPDWORD lpFileSizeHigh);
+BOOL GetFileSizeEx (HANDLE hFile, PLARGE_INTEGER lpFileSize);
+
 HANDLE FindFirstFileA (LPCSTR lpFileName, LPWIN32_FIND_DATAA lpFindFileData);
 BOOL FindNextFileA (HANDLE hFindFile, LPWIN32_FIND_DATAA lpFindFileData);
 BOOL FindClose (HANDLE hFindFile);
 
+BOOL DeleteFileA (LPCTSTR lpFileName);
+BOOL RemoveDirectoryA (LPCSTR lpPathName);
+BOOL CreateDirectoryA (LPCSTR lpPathName, LPSECURITY_ATTRIBUTES lpSecurityAttributes);
+BOOL MoveFileA (LPCTSTR lpExistingFileName, LPCTSTR lpNewFileName);
+
 #ifndef UNICODE
+#define CreateFile CreateFileA
 #define FindFirstFile FindFirstFileA
 #define FindNextFile FindNextFileA
+static BOOL DeleteFile (LPCTSTR lpFileName) { return DeleteFileA(lpFileName); }
```

Comment:
Why is this one singled out?

Reply:
Afaik Windows uses a macro for `DeleteFile` too, but for us, this causes issues. In `xboxkrnl.h`, the struct `FILE_DISPOSITION_INFORMATION` has a field named `DeleteFile`, which was affected when including `fileapi.h` before `xboxkrnl.h`.

My original solution added a workaround to `xboxkrnl.h` using `#pragma` to store away the define, undefine it, and restore it at the end. @mborgerson suggested to use a static function instead to escape the whole macro issue, and after thinking about it for a while I decided to follow the suggestion.

As for why I didn't change the rest: Rebase conflicts. There are several commits touching these lines, and even that change caused a lot of conflicts that I had to resolve manually - so if we want to change all of them to static functions, I'd rather do that in a follow-up PR tbh.

The whole discussion about it happened in Discord, starting here: https://discordapp.com/channels/428359196719972353/428359408204906497/607000885716975628

Reply:
```c
#define DeleteFile(...) DeleteFileA(__VA_ARGS__)
```

is it possible to use a function macro like this?

I'm fine with changing rest in a follow-up. Ideally we'd provide a function alias or something.

We should also check how Microsoft did it for something like the XDK - because they'd have had the same issue.

Reply:
Well, now I'm embarrassed I didn't think of that.
It in fact solves the conflict while argument type and count checks still work fine. I'll push an update soon.

I don't know about XDK, but we dug up a [MS header](https://github.com/microsoft/service-fabric/blob/6f2a641bdfe4b7e7100780cd4a4473f8eda30d5c/src/prod/ktl/src/linuxinc/ktlpal.h#L635-L638) which simply undefined the macro, while `ntddk.h` seems to just ignore the issue (which you may get away with depending on your include order).
I don't know how they solve the problem in their code files however - in our implementation of `DeleteFileA`, we need to access the `DeleteFile`-field, so we had the conflict there, too. They must've had the same issue.

Reply:
I took a bit longer, but I just pushed an update that changes this (and all the defines in the commits that follow) to a variadic macro.